### ISSUE TITLE
Ebean13.25.2-jakarta generate unsupport DDL SQL on Sqlite3

### ISSUE THREAD

**Issue by zacker330** [ebean-orm/ebean#3312]
## Expected behavior
Generate DDL SQL that the Sqlite3 support on the unidirectional relationship with `@OneToOne` and `@JoinColumn`

## Actual behavior
Generate the sql like following that Sqlite3 don't support.

```sql
alter table ossl_artifact_versioning add constraint fk_ossl_artifact_versioning_coor_id foreign key (coor_id) references ossl_artifactcoordinates (id) on delete restrict on update restrict;
```

### Steps to reproduce

```java
// in Artifact class
@OneToOne(fetch = FetchType.EAGER)
@JoinColumn(name = "coor_id",referencedColumnName = "id")
private ArtifactCoordinate artifactCoordinate;

// in ArtifactCoordinate class
@Id
private long id;

// none fields assosiate with Artifact class

```

The logs as following I got after I run the generation and ddl scripts at the condition that the migrations folder and dbinit folder is empty:

```console
08:42:24.834 [main] INFO  io.ebean - ebean version: 13.25.2-jakarta
08:42:24.870 [main] INFO  io.avaje.config - Loaded properties from [] 
08:42:25.996 [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Starting...
08:42:27.315 [main] INFO  com.zaxxer.hikari.pool.HikariPool - HikariPool-1 - Added connection org.sqlite.jdbc4.JDBC4Connection@563f38c4
08:42:27.324 [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Start completed.
08:42:27.348 [main] WARN  io.ebean.core - DataSource [domain-sql] has autoCommit defaulting to true!
08:42:31.007 [main] INFO  io.ebean.DDL - Executing domain-sql-drop-all.sql - 8 statements, autoCommit:true
08:42:31.025 [main] INFO  io.ebean.DDL - Executing domain-sql-create-all.sql - 8 statements, autoCommit:false
Exception in thread "main" jakarta.persistence.PersistenceException: Failed to run script
        at io.ebeaninternal.dbmigration.DdlGenerator.runScript(DdlGenerator.java:171)
        at io.ebeaninternal.dbmigration.DdlGenerator.runCreateSql(DdlGenerator.java:201)
        at io.ebeaninternal.dbmigration.DdlGenerator.runDdlWith(DdlGenerator.java:128)
        at io.ebeaninternal.dbmigration.DdlGenerator.runDdl(DdlGenerator.java:113)
        at io.ebeaninternal.dbmigration.DdlGenerator.execute(DdlGenerator.java:89)
        at io.ebeaninternal.server.core.DefaultServer.executePlugins(DefaultServer.java:193)
        at io.ebeaninternal.server.core.DefaultContainer.startServer(DefaultContainer.java:157)
        at io.ebeaninternal.server.core.DefaultContainer.createServer(DefaultContainer.java:122)
        at io.ebeaninternal.server.core.DefaultContainer.createServer(DefaultContainer.java:33)
        at io.ebean.DatabaseFactory.createInternal(DatabaseFactory.java:135)
        at io.ebean.DatabaseFactory.create(DatabaseFactory.java:84)
        at internal.DdlGeneratorCommand.main(DdlGeneratorCommand.java:88)
Caused by: java.sql.SQLException: Error executing [alter table ossl_artifact_versioning add constraint fk_ossl_artifact_versioning_coor_id foreign key (coor_id) references ossl_artifactcoordinates (id) on delete restrict on update restrict] error[[SQLITE_ERROR] SQL error or missing database (near "constraint": syntax error)]
        at io.ebean.ddlrunner.DdlRunner.runStatement(DdlRunner.java:106)
        at io.ebean.ddlrunner.DdlRunner.runStatements(DdlRunner.java:72)
        at io.ebean.ddlrunner.DdlRunner.runAll(DdlRunner.java:55)
        at io.ebeaninternal.dbmigration.DdlGenerator.runScript(DdlGenerator.java:163)
        ... 11 more
Caused by: org.sqlite.SQLiteException: [SQLITE_ERROR] SQL error or missing database (near "constraint": syntax error)
        at org.sqlite.core.DB.newSQLException(DB.java:941)
        at org.sqlite.core.DB.newSQLException(DB.java:953)
        at org.sqlite.core.DB.throwex(DB.java:918)
        at org.sqlite.core.NativeDB.prepare_utf8(Native Method)
        at org.sqlite.core.NativeDB.prepare(NativeDB.java:134)
        at org.sqlite.core.DB.prepare(DB.java:257)
        at org.sqlite.jdbc3.JDBC3Statement.execute(JDBC3Statement.java:52)
        at com.zaxxer.hikari.pool.ProxyStatement.execute(ProxyStatement.java:94)
        at com.zaxxer.hikari.pool.HikariProxyStatement.execute(HikariProxyStatement.java)
        at io.ebean.ddlrunner.DdlRunner.runStatement(DdlRunner.java:101)
        ... 14 more
make: *** [ddl] Error 1

```

## The SQL Generated
```sql
create table ossl_artifactcoordinates (
  id                            bigint auto_increment not null,
  group_id                      varchar(255),
  artifact_id                   varchar(255),
  artifact_version              varchar(255),
  constraint pk_ossl_artifactcoordinates primary key (id)
);
create table ossl_artifact_versioning (
  repo_id                       bigint not null,
  group_id                      varchar(255),
  artifact_id                   varchar(255),
  artifact_version              varchar(255),
  scala_version                 varchar(255),
  real_artifact_id              varchar(255),
  description                   varchar(255),
  packaging                     varchar(255),
  size                          bigint not null,
  ext                           varchar(255),
  is_snapshot                   boolean default false not null,
  release_time                  timestamp,
  name                          varchar(64),
  usage                         bigint not null,
  coor_id                       bigint,
  create_time                   timestamp not null,
  when_modified                 timestamp not null,
  constraint uq_ossl_artifact_versioning_coor_id unique (coor_id),
  constraint oav_unique_group_artifact_version unique (group_id,artifact_id,artifact_version,scala_version)
);

-- foreign keys and indices
alter table ossl_artifact_versioning add constraint fk_ossl_artifact_versioning_coor_id foreign key (coor_id) references ossl_artifactcoordinates (id) on delete restrict on update restrict;




```

## What's root cause?
Accordding to the [sqlite3 doc](https://www.sqlite.org/omitted.html ) and the[ stackoverflow's answer](https://stackoverflow.com/a/1884841), is it would be correct that adding  `FOREIGN KEY`  in the create table statement than the alert table statement?

**Comment by rbygrave** [comment]
Does not reproduce:

```java
@Entity
public class Artifact {

  @Id
  long id;

  @OneToOne(fetch = FetchType.EAGER)
  @JoinColumn(name = "coor_id",referencedColumnName = "id")
  ArtifactCoordinate artifactCoordinate;

}

@Entity
public class ArtifactCoordinate {

  @Id
  long id;

  String something;
}
```

```sql
-- apply changes
create table artifact (
  id                            integer not null,
  coor_id                       integer,
  constraint uq_artifact_coor_id unique (coor_id),
  constraint pk_artifact primary key (id),
  foreign key (coor_id) references artifact_coordinate (id) on delete restrict on update restrict
);

create table artifact_coordinate (
  id                            integer not null,
  something                     varchar(255),
  constraint pk_artifact_coordinate primary key (id)
);
```

----
Now, you need to confirm that the SQLITE platform is getting detected and in the ebean-ddl-generator the `io.ebeaninternal.dbmigration.ddlgeneration.platform.SQLiteDdl` is being used.  This has `inlineForeignKeys = true;` and this is the flag that is used at: https://github.com/ebean-orm/ebean/blob/master/ebean-ddl-generator/src/main/java/io/ebeaninternal/dbmigration/ddlgeneration/platform/BaseTableDdl.java#L207-L209

So:
- confirm your maven dependency of ebean-ddl-generator is also 13.25.2 
- Put a breakpoint onto `SQLiteDdl` constructor to confirm it is being used
- Put a breakpoint onto `BaseTableDdl` line 207 to confirm it is going into there to write the foreign key inline with the create table ddl

**Comment by zacker330** [comment]
Thanks for your efforts. I'd tried to figure it out myself after your commend. 

The issue is caused by the migration code I wrote in my bazel custom rule [code](https://github.com/zacker330/bazel-springboot-project-template/blob/cca823f3368d9ef8cd026e47963e53e5279a4cf2/tools/rules_ebean/src/main/java/internal/DdlGeneratorCommand.java#L81). I didn't notice that I passed a empty platform parameter to the DatabaseConfig object
